akka-distributed-data/src/main/scala/akka/cluster/ddata/Replicator.scala

```diff
@@ -893,6 +893,22 @@ final class Replicator(settings: ReplicatorSettings) extends Actor with ActorLog
   val newSubscribers = new mutable.HashMap[String, mutable.Set[ActorRef]] with mutable.MultiMap[String, ActorRef]
   var subscriptionKeys = Map.empty[String, KeyR]
 
+  // To be able to do efficient stashing we use this field instead of sender().
+  // Using internal buffer instead of Stash to avoid the overhead of the Stash mailbox.
+  // sender() and forward must not be used from normalReceive.
+  // It's set to sender() from aroundReceive, but is also used when unstashing
+  // messages after loading durable data.
+  var replyTo: ActorRef = null
```

Comment:
Why is this a global var? Can't we add a `replyTo` to the signature of every call and then change  normalReceive to
```
val normalReceive: Receive = { case x => normalReceiveWithSender(sender())}
def normalReceiveWithSender(replyTo: ActorRef): Receive => ....
```

and then instead of changing the global var in line 955 change line 956 to:
```
normalReceiveWithSender(snd).applyOrElse(msg, unhandled)
```

Is there a reason I'm missing. I think putting it in the function signature will make it more clear what is happening.